src/language-server/language-server.ts

```diff
@@ -70,8 +80,29 @@ export async function startLanguageServer() {
     }),
   )
 
+  // Initialize PuyaService and wait for it to be ready
+  const initializePuyaService = async (): Promise<PuyaService | undefined> => {
```

Comment:
This function could take a `workspaceFolder`, then you wouldn't need to run another guard inside.